crates/lang/codegen/src/generator/storage.rs

```diff
@@ -83,6 +85,57 @@ impl Storage<'_> {
         }
     }
 
+    /// Generates a default implementation of `SpreadAllocate` for the ink! storage struct.
+    ///
+    /// # Note
+    ///
+    /// Unlike with `SpreadLayout` it is unfortunately not possible to use the derive macro
+    /// in order to implement this trait. This is due to the fact that `SpreadAllocate` shall
+    /// only be implemented on the ink! storage struct if and only if all of its fields
+    /// implement the trait.
+    /// Therefore having `SpreadAllocate` implemented on the ink! storage struct is optional
+    /// whereas having `SpreadLayout` implemented is mandatory.
+    fn generate_spread_allocate_trait_impl(&self) -> TokenStream2 {
+        let storage = self.contract.module().storage();
+        let span = storage.span();
+        let storage_ident = storage.ident();
+        let bounds = storage.fields().map(|field| {
+            let field_span = field.span();
+            let field_ty = &field.ty;
+            quote_spanned!(field_span=>
+                #field_ty: ::ink_storage::traits::SpreadAllocate
```

Comment:
I think you will have a compilation error in case of two fields with the same type

Reply:
Since `trivial_bounds` Rust feature is still unstable we have to remove this auto generated trait impl anyways unfortunately. Otherwise I guess it would be fine.